test/structs_test.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+require_relative './test_helper'
+
+class CertsTest < Minitest::Test
+  def test_changing_pem_changes_cert_properties
+    key = R509::PrivateKey.new(type: 'RSA', bit_length: 2048)
+    expired_at = Time.now.utc.to_i - 3600
+    expired_pem = cert_pem(key, 'expired.test', not_before: expired_at - 3600, not_after: expired_at)
+
+    cert = Puma::Acme::Cert.new(algorithm: :rsa, identifiers: %w[expired.test], cert_pem: expired_pem)
+    assert cert.expired?
+
+
+    unexpired_at = Time.now.utc.to_i
+    unexpired_pem = cert_pem(key, 'unexpired.test', not_before: unexpired_at - 3600, not_after: unexpired_at + 3600)
+
+    cert.cert_pem = unexpired_pem
+    assert !cert.expired?
```

Comment:
In addition to checking it is not expired any more, should we check that the x509 data is what we would expect more broadly as well? I just wouldn't want it to allow a regression where somehow expired changes but not other stuff (as unlikely as that might be).